aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3422,16 +3427,40 @@ static void linalg_lstsq_out_info(
     TORCH_INTERNAL_ASSERT(singular_values.sizes().equals(singular_values_shape));
     TORCH_INTERNAL_ASSERT(singular_values.is_contiguous());
   }
-
   // 'input' is modified in-place so we need a column-major copy
   auto input_working_copy = copyBatchedColumnMajor(input);
-
   // now the actual call that computes the result in-place (apply_lstsq)
-  lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
-
+  if(driver == "gelsd" || driver == "gelss") {
+    auto k = std::min(m, n);
```

Comment:
So the way it's setup now is that if a user specifies `gelsd` or `gelss` as their driver(both use SVD) then it'll go through the SVD implementation. With that being said,

1) This isn't a fallback mechanism. This just merely provides the user the ability to do a lstsq on rank deficient matrices with SVD. 
I do not believe this is ideal. If we want to use a fallback mechanism, then ideally, we would check `input` or `A` tensor to see if it is rank deficient. However, I do not know how to do this in a performant way, so hopefully someone can point me in the right direction. 

If you do `lstsq_stub`, errors will be reflected in `_linalg_check_errors` after the computation. I would like to avoid having to do this and figure out from the jump so we can avoid the redundant work.

Reply:
It's not possible to do detect this by default, that's why we ask the user to define the driver themselves.